pkg/cmd/provisioning/ibmcloud/refresh-keys.go

```diff
@@ -0,0 +1,159 @@
+	}
+
+	cs, err := newClientset(Options.KubeConfigFile)
+	if err != nil {
+		return errors.Wrap(err, "Failed to create the kubernetes clientset")
+	}
+	err = refreshKeys(ibmclient, cs, apiKeyDetails.AccountID, Options.Name, Options.ResourceGroupName, Options.CredRequestDir, Options.Create)
+	if err != nil {
+		return errors.Wrap(err, "Failed to refresh keys")
+	}
+	return nil
+}
+
+func refreshKeys(client ibmcloud.Client, cs *kubernetes.Clientset, accountID *string, name, resourceGroupName, credReqDir string, create bool) error {
+	resourceGroupID, err := getResourceGroupID(client, resourceGroupName)
+	if err != nil {
+		return errors.Wrap(err, "Failed to getResourceGroupID")
+	}
+
+	// Process directory
+	credReqs, err := getListOfCredentialsRequests(credReqDir)
+	if err != nil {
+		return errors.Wrap(err, "Failed to process files containing CredentialsRequests")
+	}
+
+	var serviceIDs []*ServiceID
+	for _, cr := range credReqs {
+		serviceID := NewServiceID(client, name, *accountID, resourceGroupID, cr)
+		serviceIDs = append(serviceIDs, serviceID)
+	}
+
+	for _, serviceID := range serviceIDs {
+		exists, list, err := serviceID.Exists()
+		if err != nil {
+			return errors.Wrapf(err, "Failed to check an existance for the ServiceID: %s", serviceID.name)
+		}
+		if !exists && !create {
+			return fmt.Errorf("ServiceID: %s does not exist, rerun with --create flag to create it", serviceID.name)
+		}
+		if len(list) > 1 {
+			return fmt.Errorf("more than one ServiceID found with %s name, please delete the duplicate entries", serviceID.name)
+		}
+	}
+
+	for _, serviceID := range serviceIDs {
+		log.Printf("Refershing the token for ServiceID: %s", serviceID.name)
+		exists, list, err := serviceID.Exists()
+		if err != nil {
+			return errors.Wrapf(err, "Failed to check an existance for the ServiceID: %s", serviceID.name)
+		}
+		if exists {
+			serviceID.ServiceID = &list[0]
+			if err := serviceID.Refresh(); err != nil {
+				return errors.Wrapf(err, "Failed to create API Key for ServiceID: %s", serviceID.name)
+			}
+		} else {
+			log.Printf("ServiceID does not exist, creating it.")
+			if err := serviceID.Do(); err != nil {
+				return errors.Wrap(err, "Failed to process the serviceID")
+			}
+		}
+
+		secret, err := serviceID.DumpAsSecret()
+		if err != nil {
+			return errors.Wrapf(err, "Failed to Dump the secret for the serviceID: %s", serviceID.name)
+		}
+		data, err := json.Marshal(secret)
+		if err != nil {
+			return errors.Wrapf(err, "Failed to Marshal")
+		}
+
+		log.Printf("Updating/Creating the secret: %s/%s for the serviceID: %s", secret.Namespace, secret.Name, serviceID.name)
+		//TODO(mkumatag): replace Patch() call with Apply() after k8s.io/client-go update to v1.21.0 or later
+		if _, err := cs.CoreV1().Secrets(secret.Namespace).Patch(context.TODO(), secret.Name, types.ApplyPatchType, data, metav1.PatchOptions{FieldManager: "cloud-credential-operator"}); err != nil {
```

Comment:
I'd prefer to mark it as owned by `ccoctl` to make it clear where this Secret came from. Or is there some expectation that the Manager field be set to be associated with the in-cluster operator?
```suggestion
		if _, err := cs.CoreV1().Secrets(secret.Namespace).Patch(context.TODO(), secret.Name, types.ApplyPatchType, data, metav1.PatchOptions{FieldManager: "ccoctl"}); err != nil {
```